a8c-files.php

```diff
@@ -490,24 +504,55 @@ function delete_file( $file_name ) {
 	private function purge_file_cache( $url, $method ) {
 		global $file_cache_servers;
 
+		$parsed = parse_url( $url );
+		if ( empty( $parsed['host'] ) ) {
+			return $requests;
+		}
+
+		$uri = '/';
+		if ( isset( $parsed['path'] ) )
+			$uri = $parsed['path'];
+		if ( isset( $parsed['query'] ) )
+			$uri .= $parsed['query'];
+
 		$requests = array();
 
-		if ( ! isset( $file_cache_servers ) || empty( $file_cache_servers ) )
-			return $requests;
+		if ( defined( 'PURGE_SERVER_TYPE' ) && 'mangle' == PURGE_SERVER_TYPE ) {
+			$data = array(
+				'group' => 'vip-go',
+				'scope' => 'global',
+				'type'  => $method,
+				'uri'   => $parsed['host'] . $uri,
+				'cb'    => 'nil',
+			);
+			$json = json_encode( $data );
 
-		$parsed = parse_url( $url );
-		if ( empty( $parsed['host'] ) )
+			$curl = curl_init();
+			curl_setopt( $curl, CURLOPT_URL, constant( 'PURGE_SERVER_URL' ) );
+			curl_setopt( $curl, CURLOPT_POST, true );
+			curl_setopt( $curl, CURLOPT_POSTFIELDS, $json );
+			curl_setopt( $curl, CURLOPT_HTTPHEADER, array(
+					'Content-Type: application/json',
+					'Content-Length: ' . strlen( $json )
+				) );
+			curl_setopt( $curl, CURLOPT_TIMEOUT, 5 );
+			curl_exec( $curl );
+			$http_code = curl_getinfo( $curl, CURLINFO_HTTP_CODE );
+			curl_close( $curl );
+
+			if ( 200 != $http_code ) {
+				error_log( sprintf( __( 'Error purging %s from the cache service: Code %d' ), $url, $http_code ) );
```

Comment:
To ensure the error actually gets indexed: 

```suggestion
				trigger_error( sprintf( __( 'Error purging %s from the cache service: Code %d' ), $url, $http_code ), E_USER_WARNING );
```